lib/src/ostree_manual.rs

```diff
@@ -0,0 +1,35 @@
+//! Manual workarounds for ostree bugs
+
+use std::io::Read;
+use std::ptr;
+
+use ostree;
+use ostree::prelude::{Cast, InputStreamExtManual};
+use ostree::{gio, glib};
+
+#[allow(unsafe_code)]
+
+/// Equivalent of `g_file_read()` for ostree::RepoFile to work around an ostree bug.
+pub fn repo_file_read(f: &ostree::RepoFile) -> Result<gio::InputStream, glib::Error> {
+    use glib::translate::*;
+    let stream = unsafe {
+        let f = f.upcast_ref::<gio::File>();
+        let mut error = ptr::null_mut();
+        let stream = gio::ffi::g_file_read(f.to_glib_none().0, ptr::null_mut(), &mut error);
+        if !error.is_null() {
+            return Err(from_glib_full(error));
+        }
+        let stream = stream as *mut gio::ffi::GInputStream;
```

Comment:
It would be good to make sure that `stream` is non-null before doing this.

Reply:
We're just casting the pointer, not dereferencing here.  The dereference happens in `from_glib_full`, which already has a null check: https://docs.rs/glib/0.14.4/src/glib/shared.rs.html#422